Add optional fee_per_gram field to console wallet Send Tx form

philipr-za opened the pull request:
## Description
This PR adds an optional editable fee_per_gram field to the send transaction form in the console wallet.

This field is populated by the default value specified by a constant in LibWallet and is not included in the flow of fields edited while sending a transaction as editing this field should be very rare. If the user wants to edit the field they can use the hotkey and after this transaction is sent the field will be repopulated with the default value again.

## How Has This Been Tested?

This functionality was not able to be tested as the network is currently experiencing some issues that make sending transactions difficult.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I ran `cargo test` successfully before submitting my PR.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

philipr-za commented:
Added a quick update to not quit on a plain 'q' but to quit on ctrl+q and ctrl+c

delta1 reviewed (approved):
clippy fail is from #2413

delta1 reviewed (changes requested):
just needs a clippy fix then happy to approve

philipr-za committed:
Add optional fee_per_gram field to console wallet Send Tx form

This PR adds an optional editable fee_per_gram field to the send transaction form in the console wallet.

This field is populated by the default value specified by a constant in LibWallet and is not included in the flow of fields edited while sending a transaction as editing this field should be very rare. If the user wants to edit the field they can use the hotkey and after this transaction is sent the field will be repopulated with the default value again.

This functionality was not able to be tested as the network is currently experiencing some issues that make sending transactions difficult.

philipr-za commented:
Fixed, though issue came from development. Thou shalt appease clippy!

delta1 commented:
yeah I blame @mikethetike 😆

delta1 reviewed (approved):
LGTM

StriderDM commented on base_layer/wallet/src/types.rs:
```diff
@@ -20,8 +20,13 @@
 
+/// The default fee per gram that the wallet will use to build transactions.
+/// TODO discuss what the default fee value should actually be
+pub const DEFAULT_FEE_PER_GRAM: MicroTari = MicroTari(25);
```
Minimum fee in mobile wallet is set to 100.

Not sure if these should be in sync?

philipr-za replied:
This is fee_per_gram not absolute fee. This is also an arb selection and needs to be discussed as stated in the TODO
